packages/react-composites/tests/browser/chat/ChatComposite.test.ts

```diff
@@ -127,10 +127,11 @@ test.describe('Chat Composite custom data model', () => {
     await page.type(dataUiId(IDS.sendboxTextfield), 'How the turn tables');
     await page.keyboard.press('Enter');
     await page.waitForSelector(`[data-ui-status="delivered"]`);
-
+    // Now that we are using a promise based custom avatar data injection, we
+    // need to wait for the avatar to load before we can take a screenshot.
+    await page.waitForTimeout(1000);
```

Comment:
This adds 1 second to the test unconditionally.

Instead of this, can you inject and wait for an HTML attribute here as well? (e.g., inject a more specific test in the `text: 'CustomDataHookTextWithMagicNumber23997'`` and then wait for an HTML element with that text).

Reply:
Can't do that here since we are not injecting an html element, rather injecting data values to the avatar.